src/tex/optimize.cpp

```diff
@@ -171,6 +209,45 @@ auto compute_optimization_steps(const Texture &file, const Settings &sets) noexc
     return res;
 }
 
+auto compute_optimization_steps(const CrunchTexture &file, const Settings &sets) noexcept -> OptimizationSteps
+{
+    const auto &tex = file.get();
+
+    auto res = OptimizationSteps{};
+
+    // Resizing.
+    const auto dim = file.get_dimension();
+    const std::optional<Dimension> target_dim
+        = std::visit(common::Overload{
+                         [](std::monostate) -> std::optional<Dimension> { return {}; },
+                         [&](util::ResizeRatio r) { return std::optional(compute_resize_dimension(dim, r)); },
+                         [&](Dimension target) {
+                             return std::optional(util::compute_resize_dimension(dim, target));
+                         },
+                     },
+                     sets.resize);
+
+    if (target_dim.has_value() && dim != target_dim.value())
+        res.resize = target_dim.value();
+
```

Comment:
Please factor out common parts such as this one

Reply:
Refactored the target dimension calculation out of this, don't think there is too much else to do. 🤔